src/js/common/components/Modal.jsx

```diff
@@ -32,8 +35,10 @@ class Modal extends React.Component {
 
   componentWillReceiveProps(newProps) {
     if (newProps.visible && !this.props.visible) {
+      document.addEventListener('keyup', this.handleDocumentKeyUp, false);
       this.setState({ lastFocus: document.activeElement, focusListener: focusListener(newProps.focusSelector) });
     } else if (!newProps.visible && this.props.visible) {
+      document.removeEventListener('click', this.handleDocumentKeyUp, false);
```

Comment:
Isn't `keyup` the right event to unbind the listener from rather than `click`?

Reply:
Yeah it definitely is. Wow, this is sloppiest PR I've ever submitted. Sorry man, thanks for the thorough CR.